search/src/search.rs

```diff
@@ -1,13 +1,38 @@
 use std::collections::{HashMap, LinkedList};
 use tantivy::collector::TopDocs;
 use tantivy::query::QueryParser;
-use tantivy::{DocAddress, Index, IndexReader, ReloadPolicy, Score};
+use tantivy::{DocAddress, IndexReader, ReloadPolicy, Score, Snippet, SnippetGenerator};
 
+extern crate litt_index;
+use litt_index::index::Index;
 use litt_shared::search_schema::SearchSchema;
 
 use crate::LittSearchError::{InitError, SearchError};
 use crate::Result;
 
+#[derive(Debug, Clone, Copy)]
+pub struct SearchResult {
+    pub page: u32,
+    segment_ord: u32,
+    doc_id: u32,
+}
+
+impl SearchResult {
+    pub fn new(page: u32, segment_ord: u32, doc_id: u32) -> Result<Self> {
+        Ok(Self {
+            page,
+            segment_ord,
+            doc_id,
+        })
+    }
+}
+
+impl PartialEq for SearchResult {
+    fn eq(&self, other: &Self) -> bool {
+        self.page == other.page
+    }
```

Comment:
What is the reason that only the page number determines that search results are equal?

Reply:
The search returns the following hashmap: 
`HashMap<String, LinkedList<SearchResult>>` where String is the title of a book and SearchResult contains the page, and a document-id tuple for reconstruction. (Keep in mind that every one pdf is actually represented by n documents, where n is equal to the number of pages.) 

Thus when I want to compare two search results it is sufficient to compare only the page, especially since such a comparison is only needed for testing. 

However, since comparing the other values too shouldn't change anything, I'll and it anyway since it removes the confusion :)

Reply:
This is now fixed